File: WordPress/src/androidTest/java/org/wordpress/android/e2e/SignUpTests.java
package org.wordpress.android.e2e;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.wordpress.android.e2e.flows.SignupFlow;
import org.wordpress.android.support.BaseTest;

import static org.wordpress.android.BuildConfig.E2E_SIGNUP_DISPLAY_NAME;
import static org.wordpress.android.BuildConfig.E2E_SIGNUP_EMAIL;
import static org.wordpress.android.BuildConfig.E2E_SIGNUP_PASSWORD;
import static org.wordpress.android.BuildConfig.E2E_SIGNUP_USERNAME;

import dagger.hilt.android.testing.HiltAndroidTest;

@HiltAndroidTest
public class SignUpTests extends BaseTest {
    @Before
    public void setUp() {
        logoutIfNecessary();
    }

    @After
    public void tearDown() {
        logoutIfNecessary();
    }

    @Test
    public void e2eSignUpWithMagicLink() {
        new SignupFlow().chooseContinueWithWpCom()
                        .enterEmail(E2E_SIGNUP_EMAIL)
                        .openMagicLink()
                        .checkEpilogue(
                                E2E_SIGNUP_DISPLAY_NAME,
                                E2E_SIGNUP_USERNAME)
                        .enterPassword(E2E_SIGNUP_PASSWORD)
                        .dismissInterstitial()
                        .dismissJetpackAd()
                        .confirmSignup();
    }
}

File: WordPress/src/androidTest/java/org/wordpress/android/e2e/SignUpTests.kt
package org.wordpress.android.e2e

import dagger.hilt.android.testing.HiltAndroidTest
import org.junit.Before
import org.junit.Test
import org.wordpress.android.e2e.flows.SignupFlow
import org.wordpress.android.support.BaseTest
import org.wordpress.android.support.ComposeEspressoLink
import org.wordpress.android.support.E2ECredentials

@HiltAndroidTest
class SignUpTests : BaseTest() {
    @Before
    fun setUp() {
        ComposeEspressoLink().unregister()
        logoutIfNecessary()
    }

    @Test
    fun e2eSignUpWithMagicLink() {
        try {
            SignupFlow().chooseContinueWithWpCom(super.mComposeTestRule)
                .enterEmail(E2ECredentials.SIGNUP_EMAIL)
                .openMagicLink()
                .checkEpilogue(
                    E2ECredentials.SIGNUP_DISPLAY_NAME,
                    E2ECredentials.SIGNUP_USERNAME
                )
                .enterPassword(E2ECredentials.SIGNUP_PASSWORD)
                .dismissInterstitial()
                .dismissJetpackAd()
                .confirmSignup()
        } finally {
            logoutIfNecessary()
        }
    }
}
